File: TP1/PYTHON/TP3/main/main.py
from src.personne import Personne
from src.etudiant import Etudiant
from src.cours import Cours
from src.enseignant import Enseignant

if __name__ == "__main__":  
    # commande pour lancer le code
    # se placer dans Python/TP3 et faire : python -m main.main
    personnes = []

    etudiant1 = Etudiant("Bob", 22, "E123", 15.5, [])
    enseignant1 = Enseignant("Dupont", 40, 3000)

    personnes.append(etudiant1)
    personnes.append(enseignant1)

    for p in personnes:
        print(p.afficher_details())
    

File: TP1/JAVA/TP3/main/Main.java
package main;
import src.*;

import java.util.ArrayList;
import java.util.List;

public class Main {

    public static void main(String[] args) {

        List<Personne> personnes = new ArrayList<>();

        Etudiant etudiant1 = new Etudiant("Bob", 22, "E123", 15.5, new ArrayList<>());
        Enseignant enseignant1 = new Enseignant("Dupont", 40, 3000);

        personnes.add(etudiant1);
        personnes.add(enseignant1);

        for (Personne p : personnes) {
            p.afficherDetails();
        }
    }
}